lib/hashes.md

```diff
@@ -5,40 +5,68 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A Hash is a dictionary-like collection of unique keys and their values. They are similar to Arrays, but where an Array uses integers as its index, a Hash allows you to use any object type.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+As stated above, Hashes can hold any type of object data. For example: grades = { "Jane Doe" => 10, "Jim Doe" => 6 } 
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+They are similar in the fact that they both hold indexes of information like a library. The difference is that hashes can hold many more types of data than an array can.
+
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+you would use the command hash.[key]. An example would be: 
+
+elements = {"base"=>"foundation", "pedestal"=>"base"}
+  puts elements["base"]
+  
+  and that will bring back the result "foundation"
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+We would replace it by performing this operation:
+hash = {:item1 => 1}
+hash[:item2] = 2
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+I believe it is similar to that of an array. You would take your elements inside your hash and bundle them under another name, such as Contacts.
+You would then type in whatever command you wanted to go with it, like Contacts.delete[David] and that would remove David from out list.
```

Comment:
This is not actually looping through a Hash, but I can see why you might think that it is. I'll explain:

The collection which you're looping through is `rows`, and rows is an _Array_. See, if you had this...

If you had this...

```ruby
rows = ["Dan", "Mindy", "Clementine"]
```

... then you would probably have no hesitation in saying, "Yeah, that's an array". The **square brackets** around the collection's items are what gives it away that we're dealing with an array.

So your `rows`...

```ruby
rows = [
  {"Name"=>"John Doe", "State"=>"IA", "Candidate"=>"Rep. Smithers", "Amount"=>"$300"},
  {"Name"=>"Mary Smith", "State"=>"CA", "Candidate"=>"Sen. Nando", "Amount"=>"$1,000"},
  {"Name"=>"Sue Johnson", "State"=>"TX", "Candidate"=>"Rep. Nguyen", "Amount"=>"$200"}
]
```

... with its square brackets is an Array. The elements inside the Array are Hashes, sure; but `rows` itself is an Array.

---

Here is an example hash.

```ruby
scores = {"Tyler" => 500, "Beth" => 102, "Crystal" => 429}
```

 Try to write the code to loop over it so the program displays:

```
Tyler scored 500 points.
Beth scored 102 points.
Crystal scored 429 points.
```